src/Domain/Creators/TimeDependentOptions/ExpansionMap.hpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <string>
+#include <variant>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/TimeDependentOptions/FromVolumeFile.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Context.hpp"
+#include "Options/String.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain::creators::time_dependent_options {
+/*!
+ * \brief Class to be used as an option for initializing expansion map
+ * coefficients.
+ */
+struct ExpansionMapOptions {
+  using type = Options::Auto<ExpansionMapOptions, Options::AutoLabel::None>;
+  static std::string name() { return "ExpansionMap"; }
+  static constexpr Options::String help = {
+      "Options for a time-dependent expansion of the coordinates. Specify "
+      "'None' to not use this map."};
+
+  struct InitialValues {
+    using type =
+        std::variant<std::array<double, 3>, FromVolumeFile<names::Expansion>>;
+    static constexpr Options::String help = {
+        "Initial values for the expansion map, its velocity and "
+        "acceleration."};
+  };
+
+  struct InitialValuesOuterBoundary {
```

Comment:
Question: why doesn't "FromVolumeFile" just copy the FoT altogether, instead of essentially recreating it from values and derivatives?

Reply:
So originally I didn't do this because I was lazy and I just read in the last FoT in the volume file rather than finding the correct observation ID. However, now that I think about it, there's likely a real issue.

Imagine a scenario where I'm writing volume data at simulation time $t_0$. Whenever we write the volume data in human time, that's when we grab the FoTs, serialize them, and write them. There could be a case where, by the time the FoTs were written, a control system has updated the FoTs past $t_0$. This will then cause issues upon restart because the expiration times of the functions of time are all messed up and not what was expected. The control system (at the moment) expects to set all expiration times and to just have FoT values at $t_0$ exactly. So I think for the time being, I think this is how things need to work.

Reply:
Oh so the FoT in the volume data can be wrong? That sounds like a bug that we have to fix. It also affects interpolation of the volume data.

Reply:
No not wrong. If you write volume data at $t_0$, then the FoTs will be valid at $t_0$, which is needed for interpolation and other stuff. That's guaranteed. It's just that the FoTs could also be valid at times after $t_0$ as well. This is because of asynchronous nature of the updates of the FoTs.

Reply:
We currently retrieve the FoTs from the DataBox when volume data is written by the observer nodegroup, which is is not guaranteed to be close to the time the time step was taken. Is that ok?

Reply:
Yes it's fine. In order for the event to run at all the FoTs must be valid, so grabbing them later doesn't matter, so long as you're only interested in evaluating the FoTs at the simulation time of the event (like reading in/transforming volume data). But for initializing the FoTs, this isn't ok so we can't just copy the serialized FoTs, we have to call them at the initial time.

Reply:
What I understood from a meeting discussion :

Currently we write the serialized functions of time at the end of the observation chain, eg in `ContributeVolumeDataToWriter` right before we call `volume_file.write_volume_data()`. This means the functions of time can have data from later than necessary in them. That is, if we write at $t_0$ and the functions of time expire at $t'>t_0$, then when we actually do the write we could have function of time data for $t>t'$ written to disk. Most of the time this doesn't matter because we only want to evaluate the functions of time at $t_0$. However, in the case where we want to use this data as a "checkpoint" (e.g. starting at a different resolution), the future state of the functions of time is no longer correct. What we need to do is essentially "drop" this future state. This PR provides one implementation for dropping that future state.

Reply:
Ok that's fine and we can merge it. Would another possible implementation be to just copy the FoT object from the volume data and add a function that drops future data from it? If that has the same effect then it seems a lot more robust to me and avoids a lot of code in this PR. Again, if the decision is that this is more work than to maintain the new code in this PR then fine with me.

Reply:
This feature was historically intentionally omitted from the FoT's because the situation described above is the only case where you should legitimately be dropping times. We figured it would be safer to not allow dropping them because doing so would almost always be a bug.

Reply:
Ok wouldn't a function like `get_clone_until_time(t)` be a good implementation of this?

Reply:
That could work, but I'd have to think about it more carefully. It does get around the issue for why we originally didn't allow a `drop()` function.